spec/models/queries/days/day_query_spec.rb

```diff
@@ -0,0 +1,73 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2022 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'spec_helper'
+
+describe Queries::Days::DayQuery, type: :model do
+  let(:instance) { described_class.new }
+  let(:base_scope) { Day.default.order(date: :asc) }
+  let(:current_user) { build_stubbed(:admin) }
+
+  before do
+    login_as(current_user)
+  end
+
+  context 'without a filter' do
+    context 'as an admin' do
+      it 'is the same as getting all days' do
+        expect(instance.results.to_sql).to eql base_scope.to_sql
+      end
+    end
+
+    context 'as a non admin' do
+      let(:current_user) { build_stubbed(:user) }
+
+      it 'is the same as getting all days' do
+        expect(instance.results.to_sql).to eql base_scope.to_sql
+      end
+    end
+  end
+
+  context 'with a dates_interval filter' do
+    context 'with a "<>d" operator' do
```

Comment:
DONE 23d8392a0fdbcad52ee7a0fc75e7cb980b5bf857

Reply:
It looks like you're reimplementing the tested code in the test to count the number of days :)
maybe the test could have been simpler with fixed dates instead of relative dates.
But that's a detail, it looks good to me :+1: